### ISSUE TITLE
Parser Causes Exception

### ISSUE THREAD

**Issue by thatsddr** [nflaig/semantic-release-helm#28]
Hello everyone!

I am not sure if I am the only one to experience this, but since version 2.9.1 the release fails with the following error:
```
TypeError: this.buffer.substring is not a function
    at Lexer.getLine (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/parse/lexer.js:208:28)
    at Lexer.parseStream (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/parse/lexer.js:244:25)
    at parseStream.next (<anonymous>)
    at Lexer.parseNext (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/parse/lexer.js:226:36)
    at parseNext.next (<anonymous>)
    at Lexer.lex (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/parse/lexer.js:160:32)
    at lex.next (<anonymous>)
    at Parser.parse (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/parse/parser.js:159:20)
    at parse.next (<anonymous>)
    at Composer.compose (/home/runner/work/ott-channel-schedule-template/ott-channel-schedule-template/node_modules/yaml/dist/compose/composer.js:130:20) {
  pluginName: 'semantic-release-helm3'
}
```
I tried with node versions 18, 19, and 20.
I did some investigation and I think the problem might be in `lib/publish.js' on lines 23- 24: 

```
const chartYaml = await fsPromises.readFile(filePath);
const chart = yaml.parse(chartYaml);
``` 

I found [this](https://github.com/eemeli/yaml/issues/459) issue on GitHub that hints at the fact that readFile returns a buffer if called without a second argument, while [yaml.parse()](https://eemeli.org/yaml/#yaml-parse) expects a string.

I would be curious to hear someone else's opinion on this and knowing if anyone else had this issue

**Comment by nflaig** [comment]
I think you are correct, opened a PR to fix that. Would you be able to test the fix? I can release a rc

**Comment by nflaig** [comment]
Published [semantic-release-helm3@2.9.3-rc.0](https://www.npmjs.com/package/semantic-release-helm3/v/2.9.3-rc.0) with a fix

**Comment by thatsddr** [comment]
The project is for my organization, so I can't trigger a new release if I don't have new features, but as soon as I or one of my colleagues release something either in this or another project, we will try the version with the fix

**Comment by nflaig** [comment]
I think I will just release this before others run into the same issue. Thanks a lot for reporting and sorry that this caused issues.

**Comment by nflaig** [comment]
Published a new version [semantic-release-helm3@2.9.3](https://www.npmjs.com/package/semantic-release-helm3/v/2.9.3)
